iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -1203,15 +1203,44 @@ static int subscribeToNotifyStateIfNeeded(PMQTTTRANSPORT_HANDLE_DATA transport_d
     return result;
 }
 
+// The tokenData variable carries a string in the format "<name>=<value".
+// This function returns the length of the <name> (the portion of the string from position 0 to the last char before '=').
+// If the '=' character is not present (unlikely), the length returned is 0, which should be
+// properly handled by the calling function.
+static int getPropertyLength(const char* tokenData)
+{
+    int result = 0;
+
+    while (true)
+    {
+        if (tokenData[result] == '\0')
+        {
+            result = 0;
+            break;
+        }
+
+        if (tokenData[result] == '=')
+        {
+            break;
+        }
+
+        result++;
+    }
+
+    return result;
+}
 
 static bool isSystemProperty(const char* tokenData)
 {
     bool result = false;
     size_t propCount = sizeof(sysPropList) / sizeof(sysPropList[0]);
     size_t index = 0;
+    size_t tokenDataLength = getPropertyLength(tokenData);
+
     for (index = 0; index < propCount; index++)
```

Comment:
for (index = 0; tokenDataLength  > 0 && index < propCount; index++)

Reply:
the tokenDataLength must be compared with each property in the list we maintain in code, so this check would not address the issue.